app/routes/api/user.js

```diff
@@ -96,4 +98,34 @@ router.route('/users/:userId')
         });
 });
 
+router.route('/users/resetPassword')
```

Comment:
I'd prefer to extend the existing password reset API at the `/auth/password` route than add a whole new route. This is basically doing the same thing as the other, except that this needs to look up the user by the token rather than get it from the request.

You could add a check at the beginning of the existing password reset API for the resetToken, something like this (untested):
```
let userPromise = new Promise(function(resolve, reject) {
  if (request.user) {
    // for cases when user is authenticated
    resolve(request.user);
  } else if request.body.resetToken {
    // for cases when resetToken is passed in
    PasswordReset.findOne({resetToken: request.body.resetToken})
      .then(function(passwordReset) {        
        return User.findOne({email: passwordReset.email})
      })
      .then(function(user) {
        resolve(user);
      })
  } else {
    // bad request
    reject()
  }
});

userPromise.then(function(user) {
  // do existing password update code
  
})
.then(function() {
  // remove password reset object from DB if needed
  
})
```

If you wanted to get really fancy you could even move the whole lookup user by reset key into a model method on the User model.